tests/test_builders/xpath_util.py

```diff
@@ -19,24 +20,48 @@ def _gettext(node: Element) -> str:
     return ''.join(n.tail or '' for n in node)
 
 
+def _prettify(nodes: Iterable[Element]) -> str:
+    parts = []
+    for index, node in enumerate(nodes):
+        parts.append(f'(i={index}) ')
+        block = tostring(node, encoding='unicode', method='html')
+        parts.append(textwrap.shorten(block, width=640) + '\n')
+    return ''.join(parts)
+
+
 def check_xpath(
-    root: Element,
+    etree: ElementTree,
     filename: str | os.PathLike[str],
     xpath: str,
     check: str | re.Pattern[str] | Callable[[Sequence[Element]], None] | None,
     be_found: bool = True,
+    *,
+    min_count: int = 1,
```

Comment:
Nice additional functionality :+1: 

I found it really confusing in the past that most of the other arguments to this method -- or perhaps strictly speaking the test params that supply them -- aren't provided as keyword args.

Maybe another one for me to refactor after this is merged (it's not required for this change).

Reply:
I wanted to separate the different features but in the end I decided against to have less changes.